Tools/ssl/make_ssl_data.py

```diff
@@ -55,103 +109,178 @@ def _file_search(fname, pat):
 
 
 def parse_err_h(args):
-    """Parse err codes, e.g. ERR_LIB_X509: 11"""
-    pat = re.compile(r"#\s*define\W+ERR_LIB_(\w+)\s+(\d+)")
-    lib2errnum = {}
-    for match in _file_search(args.err_h, pat):
-        libname, num = match.groups()
-        lib2errnum[libname] = int(num)
+    """Parse error codes from include/openssl/err.h.in.
 
+    Detected lines match (up to spaces) "#define ERR_LIB_<LIBNAME> <ERRCODE>",
+    e.g., "# define ERR_LIB_NONE 1".
+    """
+    pat = re.compile(r"#\s*define\W+(ERR_LIB_(\w+))\s+(\d+)")
+    lib2errnum = {}
+    for match in file_search(args.err_h, pat):
+        macroname, libname, num = match.groups()
+        if macroname in ['ERR_LIB_OFFSET', 'ERR_LIB_MASK']:
+            # ignore: "# define ERR_LIB_OFFSET                 23L"
+            # ignore: "# define ERR_LIB_MASK                   0xFF"
+            continue
+        actual = int(num)
+        expect = lib2errnum.setdefault(libname, actual)
+        if actual != expect:
+            logger.warning("OpenSSL inconsistency for ERR_LIB_%s (%d != %d)",
+                           libname, actual, expect)
     return lib2errnum
 
 
 def parse_openssl_error_text(args):
-    """Parse error reasons, X509_R_AKID_MISMATCH"""
-    # ignore backslash line continuation for now
+    """Parse error reasons from crypto/err/openssl.txt.
+
+    Detected lines match "<LIBNAME>_R_<ERRNAME>:<ERRCODE>:<MESSAGE>",
+    e.g., "ASN1_R_ADDING_OBJECT:171:adding object". The <MESSAGE> part
+    is not stored as it will be recovered at runtime when needed.
+    """
+    # ignore backslash line continuation (placed before <MESSAGE> if present)
     pat = re.compile(r"^((\w+?)_R_(\w+)):(\d+):")
-    for match in _file_search(args.errtxt, pat):
+    seen = {}
+    for match in file_search(args.errtxt, pat):
         reason, libname, errname, num = match.groups()
-        if "_F_" in reason:
-            # ignore function codes
+        if "_F_" in reason:  # ignore function codes
+            # FEAT(picnixz): in the future, we may want to also check
+            # the consistency of the OpenSSL files with an external tool.
```

Comment:
Could you track that in an issue instead?

Reply:
Tracked in https://github.com/python/cpython/issues/132745.